### PR TITLE
fix: tolerate stale inline anchor references

### PR DESCRIPTION
**Pull request by vega113**
## Summary
- fix the inline reply / inline blip client crash caused by stale inline-anchor references in the client-side meta DOM
- tolerate missing reference anchors by appending the new inline locator instead of throwing `Item not found`
- add a focused test seam around the missing-reference decision path

## Root Cause
`ReplyManager` ultimately routes inline-anchor insertion through `BlipMetaDomImpl.insertInlineLocatorBefore()`. In the failing state, the renderer can hand that method a reference anchor that still exists in canonical order but is missing from the meta view's inline locator sequence. The old code treated that as a hard invariant and threw `IllegalArgumentException("Item not found: ...")`, which surfaced as the client-side crash.

## Verification
- `git diff --check`
- `sbt wave/compile`
- `sbt "show Test / definedTests"`
- `sbt "testOnly org.waveprotocol.wave.client.wavepanel.view.dom.BlipMetaDomImplTest"`
- `sbt "testOnly *BlipMetaDomImplTest"`

## Review Notes
Separate review lane found no blockers. Residual risks remain around possible inline-reply ordering drift if the deeper renderer/sequence desync occurs often, and the added client-package test seam is not part of a runnable discovered SBT test surface in the current build.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved reference resolution used during inline content insertion to prevent incorrect placements and improve stability when references are missing or unavailable.

* **Tests**
  * Added thorough tests covering reference resolution across normal and edge cases (missing or null references) to ensure consistent behavior.

### CODE REVIEW

File: wave/src/jakarta-test/java/org/waveprotocol/wave/client/wavepanel/view/dom/BlipMetaDomImplTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.waveprotocol.wave.client.common.util.StringSequence;
+
+public final class BlipMetaDomImplTest extends TestCase {
+
+  public void testResolveInlineLocatorReferenceKeepsKnownReference() {
+    StringSequence inlineLocators = StringSequence.of(java.util.List.of("inline-1", "inline-2"));
+
+    assertEquals(
+        "inline-2",
+        BlipMetaDomImpl.resolveInlineLocatorReference(inlineLocators, "inline-2"));
```
[the remark this answers is not in the document]
Comment (vega113) [reply wave/src/jakarta-test/java/org/waveprotocol/wave/client/wavepanel/view/dom/BlipMetaDomImplTest.java]: Fixed in a3395ed6 by moving the test into `wave/src/jakarta-test/java/org/waveprotocol/wave/client/wavepanel/view/dom/BlipMetaDomImplTest.java`, which is the included Jakarta test source set for this client package. `show JakartaTest / unmanagedSources` now includes the test, while `show Test / unmanagedSources` does not.
